### DISCUSSION TITLE
How to turn off --pretty?

### DISCUSSION THREAD

**Discussion by smprather** [TomWright/dasel#377] [Q&A] [answered]
I'm not even sure exactly what `--pretty` does. I have no way to compare because there is no `--no-pretty`. If you make a flag default=true, you have to have an off switch. Did I miss something?

Specifically, I'm trying to query a key with a string value. I would like to get the raw value (like `jq -r`), not the `'value'` with quotes which I'll have to post-strip. Not a big deal, but not as clean as it could be.

Workaround is to use `-w yaml`. Let me know if this is enhancement issue worthy.

Thanks for writing one *q to rule them all! Much needed!
--Myles

**Comment by TomWright** [comment, marked as answer]
`--pretty` allows you to turn on/off pretty printing.

JSON with pretty print (default):
```
{
  "name": "Tom"
}
```

JSON with pretty print off:
```
{"name":"Tom"}
```

Usage is `--pretty=true/false` and is documented here: https://daseldocs.tomwright.me/commands/select#flags-args

---

If you want to get a plain value without any encoding, you can use `-w -` to use the "plain" formatter, which under the hood is just go's `fmt.Sprint`. E.g. https://dasel.tomwright.me/s/0a257c1a-396e-4e75-820f-f61211695024

**Comment by smprather** [comment]
Thanks for the detailed answer @TomWright. I was going from the `dasel -h` message. I'm more familiar with the `--no-foo` way of setting flags to false. Though pretty is not what I want here. `-w -` is. Maybe you could add `-` as a formatting option to the `-h` help? To be honest, if it's not in `-h`, I'm likely not to look further (though I did search the Q&A before posting). Thanks again for the quick help!

![image](https://github.com/TomWright/dasel/assets/6503931/135e1ee0-bd9a-4af3-8428-02a058393866)

**Reply by TomWright** [reply]
I think that's reasonable. I think the --help needs a bit of a refresher for sure
